src/gitlab.js

```diff
@@ -52,28 +80,36 @@ export default class VerdaccioGitLab {
           }
         }
 
+        // Store found groups in cache
+        this._setCachedUserGroups(user, password, ownedGroups);
+        this.logger.trace('[gitlab] saving data in cache for user:', user);
+
         this.logger.debug('[gitlab] user:', user, 'authenticated, with groups:', ownedGroups);
-        cb(null, ownedGroups);
+        return cb(null, ownedGroups);
       });
-    }).
-    catch(error => {
+    }).catch(error => {
       this.logger.debug('[gitlab] error authenticating:', error.error || null);
-      if (error) { return cb(httperror[401]('personal access token invalid')); }
+      if (error) {
+        return cb(httperror[401]('personal access token invalid'));
+      }
     });
   }
 
   adduser(user: string, password: string, cb: Callback) {
     this.logger.trace('[gitlab] adduser called for user:', user);
-    cb(null, true);
+    return cb(null, true);
   }
 
   allow_access(user, _package: PackageAccess, cb: Callback) {
```

Comment:
This piece of code is improved by adding a third case: package doesn't require authentication and no authentication provided. Better error handling on the npm client